### PR TITLE
fix(ce-debug): gate hypotheses on a runnable reproduction check

### PR DESCRIPTION
**Pull request by tmchow**
`ce-debug` no longer lets hypothesis formation start on code reading alone: reproduction now has to produce a named, repeatable check that fails on the bug and passes once fixed, and that check becomes the instrument the rest of the run verifies against. Around that gate, the loop picks up discipline it was missing — a competing hypothesis named before the top one is tested, secrets redacted from all shown or committed evidence, debug instrumentation verified gone at self-review, a missing regression-test seam recorded as a finding instead of covered by a shallow test, and measure-first guidance for performance regressions (the description now also triggers on "slow").

## Validation

Full mechanical suite passes (3,658 tests). Behavioral eval via `bun run test:skill-eval-cell` on the `seat-cap` pipeline fixture against the working tree, on both Claude and Codex: both hosts ran the reproduction check red before hypothesizing, read the edited references, applied the fix, re-verified green, and emitted a correct `fixed-not-pushed` structured return. The redaction and perf paths have no fixture and were not exercised; both are additive conditions.

## Security Disclosure

No security-relevant changes. (The new redaction rule tightens handling of secrets in debug evidence; it grants no new capability.)

## Agent Disclosure

Claude Code · claude-fable-5

### CODE REVIEW

File: skills/ce-debug/SKILL.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ Default is **interactive**: investigate, run the Phase 2 fix-choice gate, then t
...
 
+## Secrets in evidence
+
+Debugging surfaces raw output constantly — command results, captured payloads, log excerpts — and this skill shows evidence to the user, writes it into findings and summaries, and commits artifacts. No secret (credential, token, auth header, connection string) appears in anything shown, written, or committed: redact it in place with `<REDACTED>`, and build reproduction commands against env vars so the credential stays in the environment rather than in the command line. If redaction removes what the diagnosis needs, say so and ask the user rather than un-redacting.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-debug/SKILL.md]: > a later instruction to redact it "in place" cannot satisfy this guarantee

Fixed in 4b96043 — the section now states the gate at command-construction time: keep credentials in env vars, and when a command's output may carry a secret, capture it to a file and surface only sanitized excerpts, rather than post-hoc redaction.

File: skills/ce-debug/references/investigate.md [hunk 2]
```diff
@@ -29,6 +29,8 @@ Read the **full thread**, not just the opening post — every comment, with part
 
 Confirm the bug exists and understand its behavior — run the test, trigger the error, follow the reported steps, whatever matches the input.
 
+**Reproduction is complete when you hold a named, repeatable check that fails on this bug now and will pass once it is fixed** — a test invocation, a script, a curl, a CLI run — and you have run it at least once and seen it fail on the user's exact symptom, not merely error somewhere nearby. That check is the instrument for everything downstream: Phase 2 tests hypotheses against it, and Phase 3 verifies the fix by re-running it. "I read the code and see the problem" is not reproduction. If no such check can be built here, that is the cannot-reproduce path below — document what was tried and what is missing, and ask the user for the environment, a captured artifact, or the go-ahead to proceed on observation alone; it is never license to start hypothesizing as though reproduction had succeeded.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-debug/references/investigate.md:32]: > give every non-interactive cannot-reproduce path an explicit conservative structured outcome

Fixed in 4b96043, at the owning layer: `references/pipeline-mode.md` now carries a Phase 1 override — proceed on the evidence already in reach (failing job's logs, captured artifacts, seeded log tails) without asking; a gap-free root cause takes the ordinary statuses, otherwise return `needs-human` with a `decision_context` naming what reproduction requires.

File: skills/ce-debug/references/investigation-techniques.md [hunk 3]
```diff
@@ -208,6 +208,16 @@ For test runs, most test runners integrate with the above — e.g., `node --insp
 
 ---
 
+## Performance Regressions
+
+When the symptom is "slow" rather than "wrong", logs and code reading mislead: intuition about where time goes is unreliable, and a plausible-looking hot spot is a hypothesis, not evidence. Measure first, change second:
+
+- Establish a numeric baseline before touching anything — a timing harness around the slow operation, a profiler run, a query plan (`EXPLAIN ANALYZE`). The baseline is Phase 1's reproduction check for a perf bug: the number is the red, and the fix is verified by re-measuring the same thing, not by reasoning that the change should be faster.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-debug/references/investigation-techniques.md:215]: > route performance symptoms to this reference from the always-read investigation flow

Fixed in 4b96043 — `investigate.md` §1.1 (always read for Phases 0-2) now routes performance symptoms to the Performance Regressions section before tracing, and this file's load contract at line 3 names performance regressions.

File: skills/ce-debug/SKILL.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ---
 name: ce-debug
-description: "Diagnosis loop for bugs and failing behavior. Use when asked to debug or fix failing behavior."
+description: "Diagnosis loop for bugs and failing behavior. Use when asked to debug or fix failing or slow behavior."
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-debug/SKILL.md:3]: > update the substantive guide alongside the activation change

Fixed in 4b96043 — `docs/guides/ce-debug.md` now lists performance regressions in When-to-use and explains the measurement workflow (baseline as the reproduction, re-measurement as verification).

File: skills/ce-debug/references/fix.md [hunk 5]
```diff
@@ -6,11 +6,11 @@ Read this before editing any file in Phase 3. The branch check and the pre-fix s
...
+1. Choose the regression test's home. Follow the active project instructions and any applicable subdirectory-scoped instructions, and always inspect existing tests before adding coverage. Use an existing failing test when it already captures the bug, update an existing test when it owns the contract but has the wrong expectation, strengthen an over-mocked test that should have caught the bug, or add a new minimal isolated test only when no existing test is the right home. It must fail on the current bug and pass once the corrected behavior lands; name it so the failure message explains the bug. When no available seam can exercise the bug as it actually triggered — the failure needs a chain of callers or a state the reachable seams cannot set up — do not write a shallow test there for the false confidence: record the missing seam as a finding in the debug summary, and let defense-in-depth or the post-mortem carry it. The body's precondition decides whether an existing test may be updated at all: a confirmed defect, never a test whose expectation the change deliberately reverses.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-debug/references/fix.md:9]: > make those red-green steps conditional on an available faithful test seam and define the runnable reproduction check as the verification path otherwise

Fixed in ee70648 — steps 2 and 4 are now conditional: when step 1 recorded a missing seam and wrote no test, Phase 1's reproduction check is the red-green instrument for both.

File: skills/ce-debug/references/fix.md [hunk 6]
```diff
@@ -6,11 +6,11 @@ Read this before editing any file in Phase 3. The branch check and the pre-fix s
...
 2. Verify that test fails for the right reason — the root cause, not unrelated setup.
 3. Implement the **minimal** fix: the root cause and nothing else. No drive-by refactors, formatting, or unrelated cleanup — those are separate commits.
-4. Verify the test passes, then run the broader suite for regressions.
-5. Self-review the diff — read every changed line for style violations, missed edge cases, regressions in adjacent behavior, and missing coverage. The broader polish/review/PR tail belongs to Phase 4, after the debug summary.
+4. Verify the test passes, then re-run Phase 1's reproduction check against the original scenario (not only the minimized or test-shaped one), then run the broader suite for regressions.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-debug/references/fix.md]: > Require the original-scenario rerun only when it is runnable, with a conservative terminal outcome or explicitly defined substitute evidence otherwise

Fixed in ee70648 — the re-run now applies only when the check can run here; on the pipeline cannot-reproduce path the summary or structured return states the gap and the caller's CI run on the pushed fix stands as that verification.
